crypto/onetimesig.go

```diff
@@ -37,24 +37,24 @@ type OneTimeSignature struct {
 	// Unfortunately we forgot to mark this struct as omitempty at
 	// one point, and now it's hard to change if we want to preserve
 	// encodings.
-	_struct struct{} `codec:""`
+	_struct struct{} `codec:"" vpack_assert_size:"6"`
 
 	// Sig is a signature of msg under the key PK.
-	Sig ed25519Signature `codec:"s"`
+	Sig ed25519Signature `codec:"s" vpack:"literal"`
 	PK  ed25519PublicKey `codec:"p"`
 
 	// Old-style signature that does not use proper domain separation.
 	// PKSigOld is unused; however, unfortunately we forgot to mark it
 	// `codec:omitempty` and so it appears (with zero value) in certs.
 	// This means we can't delete the field without breaking catchup.
-	PKSigOld ed25519Signature `codec:"ps"`
+	PKSigOld ed25519Signature `codec:"ps" vpack:"literal" vpack_special_values:"0"`
```

Comment:
I'm ok with it if we don't, but do we manage to eliminate this entirely, because the key and (empty) value are all one string leading right into the next string, `p2`?  Or do we end up with one byte that represents `ps=0`, followed by something for `p2`?

Reply:
I think we probably take the big win we've got and not worry about this or the other place with the 8-9 byte idea.  But nice to have the conversation here for posterity.  I suppose it matters more as stake is more decentralized - more votes per round.